examples/build_customized_agent.py

```diff
@@ -75,14 +79,15 @@ async def _act(self) -> Message:
         logger.info(f"{self._setting}: ready to {self._rc.todo}")
         todo = self._rc.todo
 
-        msg = self._rc.memory.get()[-1] # retrieve the latest memory
+        msg = self._rc.memory.get()[-1]  # retrieve the latest memory
         instruction = msg.content
 
         code_text = await SimpleWriteCode().run(instruction)
-        msg = Message(content=code_text, role=self.profile, cause_by=todo)
+        msg = Message(content=code_text, role=self.profile, cause_by=get_object_name(todo))
```

Comment:
you can define cause_by and so on as `str` type, but it's not suggested to using `get_object_name` or WriteDesign.get_class_name(). It isn't explicit for user to remember the usage and it's  repetitive (get_class_name 59 times and get_object_name 31 times in this PR). Can you improve it  with passing the raw class but using get_class_name inside Message before it's been initiated, and repeat this way inside functions like get_by_actions to deal ActionClass.

Reply:
1. The issue is a result of the design that uses classes as message filters. This design works for the standalone console version because all the information is local. However, in an instant messaging (IM) context, where message recipients are dynamically changing, it is impractical to create a class for each message recipient.

2. To convert these classes into strings, I couldn't think of a better approach.